contracts/vaults/Base.sol

```diff
@@ -290,15 +290,17 @@ contract  Base is Governable, ReentrancyGuard {
         pool.lastProcessedTotalAllocPoint = lastPoolUpdate;
     }
 
-    // Safe rewardToken transfer function, transfer HATs from the contract only if they are earmarked for rewards
+    /**
+    * @dev Safe HAT transfer function, transfer rewards from the contract only if there are enough
+    * rewards available.
+    * @param _to The address to transfer the reward to
+    * @param _amount The amount of rewards to transfer
+    * @param _pid The pool id
+   */
     function safeTransferReward(address _to, uint256 _amount, uint256 _pid) internal {
-        if (_amount > rewardAvailable) { 
-            _amount = rewardAvailable; 
-        }
-        rewardAvailable -= _amount;
+        require(rewardToken.balanceOf(address(this)) >= _amount, "HVE46");
```

Comment:
Please do not do this The balance of HATs is (a) the amount reservered for rewards and (b) the amount deposited in the vaults that hold HATs. With this change, you are paying rewards from a vault.

Reply:
but they are deposited into the vault also..
https://github.com/hats-finance/hats-contracts/blob/e3b6ae9d16a8babfa4ce647e15acf41f3e2efcf9/contracts/vaults/Deposit.sol#L23